### PR TITLE
New task

### PR DESCRIPTION
**Pull request by jspuser123**
sir pls review

### CODE REVIEW

File: coolapp/coolapp/doctype/article/article.json [hunk 1]
```diff
@@ -84,14 +78,29 @@
...
+   "in_list_view": 1,
+   "label": "issued_count",
+   "read_only": 1
+  },
+  {
+   "fieldname": "owned_book",
+   "fieldtype": "Int",
+   "in_list_view": 1,
+   "label": "owned_book",
+   "read_only_depends_on": "eval:!(doc.__islocal)",
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/article/article.json:96]: explain this to me

File: coolapp/coolapp/doctype/library_membership/library_membership.js [hunk 2]
```diff
@@ -2,7 +2,112 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Library Membership', {
-	// refresh: function(frm) {
 
-	// }
+	refresh: function(frm) 
+	{
+
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.js]: remove empty spaces

File: coolapp/coolapp/doctype/library_membership/library_membership.json [hunk 3]
```diff
@@ -18,7 +18,8 @@
   "lft",
   "rgt",
   "is_group",
-  "parent_library_membership"
+  "parent_library_membership",
+  "membership_type"
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.json]: move to top

File: coolapp/coolapp/doctype/library_membership/library_membership.py [hunk 4]
```diff
@@ -4,9 +4,48 @@
...
+
+
+
 
 class LibraryMembership(Document):
+
+
+	
 	def before_submit(self):
+
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.py]: remove unwanted empty spaces

File: coolapp/coolapp/doctype/library_membership/library_membership.py [hunk 5]
```diff
@@ -4,9 +4,48 @@
...
+		
+		if self.membership_type == 'Individual':
+			self.active_member()
+			if self.library_member and self.from_date and self.to_date and self.paid:
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.py]: - move this condition to "madatory depends on" field.
- then simplify this if conditions

File: coolapp/coolapp/doctype/library_membership/library_membership.py [hunk 6]
```diff
@@ -15,4 +54,4 @@ def before_submit(self):
             },
         )
 		if exists:
-			frappe.throw("There is an active membership for this member")
+			("There is an active membership for this member")
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.py]: fix this

File: coolapp/coolapp/doctype/library_membership/library_membership.py [hunk 7]
```diff
@@ -4,9 +4,48 @@
...
+		if self.membership_type == 'Individual':
+			self.active_member()
+			if self.library_member and self.from_date and self.to_date and self.paid:
+				print("ok")
+			else:
+				frappe.throw("Your Text Field Not Fill")
+				
+				
+			
+		elif self.membership_type == 'group':
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.py]: existing membership validation for group membership

File: coolapp/coolapp/doctype/library_transaction/library_transaction.json [hunk 8]
```diff
@@ -27,24 +29,26 @@
...
   {
+   "default": "Today",
    "fieldname": "date",
    "fieldtype": "Date",
    "in_list_view": 1,
    "label": "date",
    "reqd": 1
   },
   {
+   "depends_on": "eval:(doc.type == 'Issued' || doc.type == 'Return')",
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_transaction/library_transaction.json:46]: also add "mandatory depends on"

File: coolapp/coolapp/doctype/library_transaction/library_transaction.py [hunk 9]
```diff
@@ -7,32 +7,73 @@
 
 
 
+#import pdb; pdb.set_trace()
 class LibraryTransaction(Document):
-
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_transaction/library_transaction.py:11]: not yet reviewed. please remove unwanted empty lines.

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: coolapp/coolapp/doctype/library_transaction/library_transaction.json (hunk 8 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_transaction/library_transaction.json:46]: OK add "mandatory depends on"

File: coolapp/coolapp/doctype/library_membership/library_membership.js (hunk 2 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.js]: ok remove spaces

File: coolapp/coolapp/doctype/library_membership/library_membership.json (hunk 3 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.json]: change move to top

File: coolapp/coolapp/doctype/library_membership/library_membership.py (hunk 5 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.py]: ok add "madatory depends on" field.
then if conditions

File: coolapp/coolapp/doctype/library_membership/library_membership.py (hunk 6 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.py]: ok fix

File: coolapp/coolapp/doctype/library_membership/library_membership.py (hunk 7 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.py]: new validation add

File: coolapp/coolapp/doctype/article/article.json (hunk 1 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/article/article.json:96]: If eval:!(doc.__islocal) is true, the document is local(not saved)

local = local to your browser, i.e. not on the server

eval: = evaluate

! = not equals

doc = document

__islocal = is not saved

File: coolapp/coolapp/doctype/library_membership/library_membership.js [hunk 10]
```diff
@@ -1,8 +1,61 @@
...
 frappe.ui.form.on('Library Membership', {
-	// refresh: function(frm) {
+	refresh: function(frm) 
+		{
+			cur_frm.fields_dict['parent_library_membership'].get_query = function(doc) 
+			{
+				return {filters: {	"membership_type":'company'}}
+			}
+			frm.doc.paid=true	
+			frm.doc.is_group=true
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.js]: please fix this.

File: coolapp/coolapp/doctype/library_membership/library_membership.js [hunk 11]
```diff
@@ -1,8 +1,61 @@
...
+					}
+				}).then(lis =>
+
+				{
+				
+				if(lis.length == 0)
+					{
+					console.log("value is emty")
+					}
+				else
```
Comment (karthikeyan5) [inline coolapp/coolapp/doctype/library_membership/library_membership.js]: simplify this if condition.

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: coolapp/coolapp/doctype/library_membership/library_membership.js (hunk 10 above)
Comment (jspuser123) [reply coolapp/coolapp/doctype/library_membership/library_membership.js]: ok fix this bug

**Pull request merged**